core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala

```diff
@@ -328,7 +328,9 @@ class SparkThrowableSuite extends SparkFunSuite {
         }
       } else {
         assert(subErrorDoc.trim == errorsInDoc.trim,
-          "The error class document is not up to date. Please regenerate it.")
+          "The error class document is not up to date. Please regenerate it by running " +
+            "`SPARK_GENERATE_GOLDEN_FILES=1 build/sbt \"core/testOnly *SparkThrowableSuite -- -t " +
+            "\\\"Error classes match with document\\\"\"`")
```

Comment:
Can we use a variable of string to replace  this hardcoded instruction and put that val at the begging of this class and under the comment of how to re-generate the golden file?

The reason it is better because it is possible that the instruction of how to re-gen is updated but the developer fail to update the instruction in the such messages.

Reply:
E.g. 
```
  /* Used to regenerate the error class file. Run:
   {{{
      SPARK_GENERATE_GOLDEN_FILES=1 build/sbt \
        "core/testOnly *SparkThrowableSuite -- -t \"Error classes are correctly formatted\""
   }}}

   To regenerate the error class document. Run:
   {{{
      SPARK_GENERATE_GOLDEN_FILES=1 build/sbt \
        "core/testOnly *SparkThrowableSuite -- -t \"Error classes match with document\""
   }}}
   */
   val error_message_instrcution = Please regenerate it by running " +
            "`SPARK_GENERATE_GOLDEN_FILES=1 build/sbt \"core/testOnly *SparkThrowableSuite -- -t " +
            "\\\"Error classes match with document\\\"\"
  ```

Reply:
Sounds good. Will update

Reply:
I agree but try to avoid duplicates. Just combine a short comment + a `val` with a command.

Reply:
Updated!